packages/cactus-plugin-persistence-fabric/src/main/typescript/plugin-persistence-fabric.ts

```diff
@@ -0,0 +1,914 @@
+          }
+
+          if (checkBlock) {
+            await this.migrateBlockNrWithTransactions(blockNumber);
+            this.lastSeenBlock = Number(blockNumber);
+            if (tempBlockNumber > this.lastBlock) {
+              this.lastBlock = tempBlockNumber;
+            }
+          }
+          if (!this.synchronizationGo) {
+            checkBlock = false;
+          }
+          tempBlockNumber = tempBlockNumber + 1;
+        }
+
+        blockNumber = tempBlockNumber.toString();
+      }
+      this.lastBlock = await this.lastBlockInLedger(signingCredential);
+    } while (this.synchronizationGo);
+    return "stopped";
+  }
+
+  async changeSynchronization(): Promise<boolean> {
+    if (this.synchronizationGo) {
+      this.synchronizationGo = false;
+    } else {
+      this.synchronizationGo = true;
+    }
+    return this.synchronizationGo;
+  }
+
+  async getBlockFromLedger(blockNumber: string): Promise<any> {
+    const block: AxiosResponse<GetBlockResponseV1> = await this.apiClient.getBlockV1(
+      {
+        channelName: this.ledgerChannelName,
+        gatewayOptions: this.gatewayOptions,
+        query: {
+          blockNumber,
+        },
+        skipDecode: false,
+      },
+    );
+
+    const tempBlockParse = block.data;
+
+    return tempBlockParse;
+  }
+
+  // Migration of block nr with transaction inside
+  // NOTE that each block have at least 1 transaction endorsement
+  /**
+   *
+   * @param blockNumber this is parameter of function which set
+   * block number to be moved from ledger to database
+   * @returns true a boolean which indicates successfull migration
+   */
+  public async migrateBlockNrWithTransactions(
+    blockNumber: string,
+  ): Promise<boolean> {
+    const block: AxiosResponse<GetBlockResponseV1> = await this.apiClient.getBlockV1(
+      {
+        channelName: this.ledgerChannelName,
+        gatewayOptions: this.gatewayOptions,
+        query: {
+          blockNumber,
+        },
+      },
+    );
+
+    const tempBlockParse: any = JSON.parse(JSON.stringify(block.data));
+
+    const hash = Buffer.from(
+      tempBlockParse.decodedBlock.header.data_hash.data,
+    ).toString("hex");
+
+    const block_data = {
+      fabric_block_id: hash,
+      fabric_block_num: Number(blockNumber),
+      fabric_block_data: JSON.stringify(block.data),
+    };
+    // if (!this.dbClient.isConnected) {
+    //   await this.dbClient.connect();
+    //   this.isConnected = true;
+    // }
+
+    // Put scrapped block into database
+    const txLen = tempBlockParse.decodedBlock.data.data.length;
+    if (txLen === 0) {
+      return false;
+    }
+    for (let txIndex = 0; txIndex < txLen; txIndex++) {
+      const transactionDataObject =
+        tempBlockParse.decodedBlock.data.data[txIndex];
+      const transactionDataStringifies = JSON.stringify(transactionDataObject);
+
+      let txid = "";
+
+      let endorser_signature = "";
```

Comment:
Instead of creating all these vars differently, can we use `InsertDetailedTransactionEntryInterface` here, have all of these referenced by a single var and then use them like var_name.endorser_signature?
And for if else statements we can do something like `if (condition) ? var_name.endorser_signature = ... : var_name.endorser_signature = ""`

Reply:
insertDetailedTransactionEntry is always executed , there are transactions like config transactions and not every transaction is related to chaincodeID, but will recheck that

Reply:
it is styling concept and I used to such view as more readable at this point

Reply:
But we already have the interface in place, which has the exact same fields. We can simply create an instance for that interface and then use it with just `instance_name.property`. (Lemme know if there are any issues in doing that way, because I would agree to the readable part only if the variables are not just `""`, but in the current case, all of them are empty strings)

Reply:
@jagpreetsinghsasan Hi, made some changes in last force push by me. Please tell me if this matches what was in your idea :)